### ISSUE TITLE
FieldExport unable to export from secret to secret

### ISSUE THREAD

**Issue by csantanapr** [aws-controllers-k8s/community#1542]
**Describe the bug**
I'm working on the ACK tutorial for the EKS Workshop, and hit this problem with both RDS and MQ

Following documentation best practices, I create a password in a "protected" namespace, then create a RDS DBInstance on this namespace with a reference to the password.

Then using FieldExports to export the DBInstance endpoint, username, and password to a secret in the "app" namespace.
The endpoint and username works from DBInstance but the password from the secret used it doesn't export.

**Steps to reproduce**

Create password
```shell
kubectl create secret generic rds-eks-workshop --from-literal=password="$(date +%s | sha256sum | base64 | head -c 32)" --namespace default
```

Create DB
```yaml
apiVersion: rds.services.k8s.aws/v1alpha1
kind: DBInstance
metadata:
  name: rds-eks-workshop2
  namespace: default
spec:
  allocatedStorage: 20
  dbInstanceClass: db.t4g.micro
  dbInstanceIdentifier: rds-eks-workshop2
  engine: mysql
  engineVersion: "8.0"
  masterUsername: "admin"
  dbSubnetGroupRef: 
    from: 
      name: rds-eks-workshop
  vpcSecurityGroupRefs:
    - from: 
        name: rds-eks-workshop
  masterUserPassword:
    namespace: default
    name: rds-eks-workshop
    key: password
  dbName: catalog
```

Create Export
```yaml
apiVersion: services.k8s.aws/v1alpha1
kind: FieldExport
metadata:
  name: catalog-reader-db-password
spec:
  to:
    name: catalog-reader-db
    kind: secret
    namespace: catalog-prod
    key: password
  from:
    path: ".data.password"
    resource:
      group: ""
      kind: Secret
      name: rds-eks-workshop
```

The password not present in secret
```
kubectl get secret catalog-reader-db -o go-template='{{.data.password|base64decode}}' -n catalog-prod
```

**Expected outcome**
The key `password` to be populated in the secret `catalog-reader-db` in the namespace `catalog-prod`

**Environment**

EKS 1.23

**Comment by A-Hilaly** [comment]
Related #1417

**Comment by RedbackThomson** [comment]
This just came back onto our plate and I wanted to make a comment about this:

tl;dr: I think that `FieldExport` should not support secrets, as it could sidestep the permissions required by application workloads to access the original K8s secrets.

The controller requires a service account that has `Secret` `Get` privileges to mount the K8s secret to a custom resource Spec. Since `FieldExport` uses the same service account, as long as it has the privileges to mount it to the CR, then it would have the privileges to export it to another `Secret`. My issue is that when workloads try to mount the new secret, they may not have had access to the original secret - which contains the same value. Essentially, this means that as long as a user can create an ACK custom resource, they essentially have permissions to export secrets that they can access.

**Comment by a-hilaly** [comment]
`FieldExport` isn't designed to export data from non ACK resources + very likely we will deprecated it in favour of a more generalistic solution.
/close
